addons/sales_team/models/crm_team_member.py

```diff
@@ -0,0 +1,188 @@
+
+    @api.depends('crm_team_id')
+    def _compute_user_company_ids(self):
+        all_companies = self.env['res.company'].search([])
+        for member in self:
+            team_company = member.crm_team_id.company_id
+            if not team_company:
+                member.user_company_ids = all_companies
+            else:
+                member.user_company_ids = member.crm_team_id.company_id
+
+    @api.depends('crm_team_id')
+    def _compute_is_membership_multi(self):
+        multi_enabled = self.env['ir.config_parameter'].sudo().get_param('sales_team.membership_multi', False)
+        self.is_membership_multi = multi_enabled
+
+    @api.depends('is_membership_multi', 'active', 'user_id', 'crm_team_id')
+    def _compute_member_warning(self):
+        """ Display a warning message to warn user they are about to archive
+        other memberships. Only valid in mono-membership mode and take into
+        account only active memberships as we may keep several archived
+        memberships. """
+        if all(m.is_membership_multi for m in self):
+            self.member_warning = False
+        else:
+            active = self.filtered('active')
+            (self - active).member_warning = False
+            if not active:
+                return
+            existing = self.env['crm.team.member'].search([('user_id', 'in', active.user_id.ids)])
+            user_mapping = dict.fromkeys(existing.user_id, self.env['crm.team'])
+            for membership in existing:
+                user_mapping[membership.user_id] |= membership.crm_team_id
+
+            for member in active:
+                teams = user_mapping.get(member.user_id, self.env['crm.team'])
+                remaining = teams - (member.crm_team_id | member._origin.crm_team_id)
+                if remaining:
+                    member.member_warning = _("Adding %(user_name)s in this team would remove him/her from its current teams %(team_names)s.",
+                                              user_name=member.user_id.name,
+                                              team_names=", ".join(remaining.mapped('name'))
+                                             )
+                else:
+                    member.member_warning = False
+
+
+    @api.model_create_multi
+    def create(self, values_list):
+        """ Specific behavior implemented on create
+
+          * mono membership mode: other user memberships are automatically
+            archived (a warning already told it in form view);
+          * creating a membership already existing as archived: do nothing as
+            people can manage them from specific menu "Members";
+        """
+        is_membership_multi = self.env['ir.config_parameter'].sudo().get_param('sales_team.membership_multi', False)
+        if is_membership_multi:
+            return super(CrmTeamMember, self).create(values_list)
+
+        self._synchronize_memberships(values_list)
```

Comment:
A bit more simple to avoid multiple return points:
```py
if not is_membership_multi:
    self._synchronize_memberships(values_list)
return super(CrmTeamMember, self).create(values_list)
```